packages/store/addon/-private/ts-interfaces/minimum-serializer-interface.ts

```diff
@@ -0,0 +1,280 @@
+/**
+  ## Overview
+
+  In order to properly manage and present your data, `EmberData`
+  needs to understand the structure of data it receives.
+
+  `Serializers` convert data between the format used by an API
+  and the format `EmberData` understands.
+
+  Data received from an API response is `"normalized"` into 
+  [JSON:API](https://jsonapi.org/) (the format used internally
+  by `EmberData`), while data sent to an API is `"serialized"`
+  into the format the API expects.
+
+  ### Implementing a Serializer
+
+  There are only two required serializer methods, one for
+  normalizing data in `JSON:API`, and another for serializing
+  records via `Snapshot`s into the expected API.
+
+  To implement a serializer, export a class implementing the
+  [MinimumSerializerInterface](MinimumSerializerInterface) from
+  the `app/serializers/` directory. An example is below.
+
+  ```ts
+  import EmberObject from '@ember/object';
+
+  export default class ApplicationSerializer extends EmberObject {
+    normalizeResponse(_, __, rawPayload) {
+      return rawPayload;
+    }
+    serialize(snapshot, options) {
+      const serializedResource = {
+        id: snapshot.id(),
+        type: snapshot.modelName,
+        attributes: snapshot.attributes()
+      };
+
+      return serializedResource;
+    }
+  }
+ ```
+
+
+  #### Serializer Resolution
+
+  The instances of serializers defined in `app/serializers/` can be looked up
+  via `store.serializerFor(name)`.
+
+  It is recommended that applications define only a single `application` adapter and serializer
+  where possible.
+
+  `serializerFor` first attempts to find a serializer with an exact match on `name`,
+  then falls back to checking for the presence of a serializer named `application`.
+
+  Because most requests in `ember-data` occur from the perspective of a primary `type`
```

Comment:
The ordering of this paragraph seems slightly odd juxtaposed with the one just before it. This explains the reasoning of what ember-data does, maybe it should actually be first?

Reply:
I can see the ordering either way “what a serializer does” vs “how/when ember-data is going to use your serializer”. I felt introducing the concept was a slightly better way to go because telling someone how a thing is used when they don’t know what that thing is seemed a bit confusing.